content/blog/validating-smi-conformance-with-meshery.md

```diff
@@ -0,0 +1,70 @@
+
+![Meshery](/img/blog/validating-smi-conformance-with-meshery/meshery-logo-light-text-side.svg#center)
+
+## SMI Conformance Tool: Meshery
+
+Meshery, the service mesh management plane, is service mesh agnostic, which is an important characteristic as a compliance tool. No better tool for the job (in fact, no other tool capable of the job) than Meshery, which has been [a partner of SMI](https://meshery.io/blog/a-standard-interface-for-service-meshes) since its inception.
+
+![SMI Conformance Tool](/img/blog/validating-smi-conformance-with-meshery/meshery-service-mesh-compliance-tool.png).
+
+Meshery is ideal tooling in that it provides lifecycle management of a large number of service meshes and sample applications which need to be provisioned, tested, and deprovisioned in the process of validating conformance. And that Meshery is capable of generating load and verifying test assertions. Meshery will be extended to validate any service mesh that claims to implement and conform SMI specifications. And, more broadly, to leverage the same tooling to validate service mesh standards at-large, including SMI and [Service Mesh Performance](https://smp-spec.io) (SMP).
+
+## Defining Conformance
+
+Conformance of SMI specifications is defined as a series of test assertions. A test assertion is a condition that must be positively verified in order for an implementation to be considered conformant. A test assertion may involve any number of conditions. Sets of test assertions are categorized by SMI specification. Collectively, these test sets comprise the complete suite of SMI conformance tests. Just like SMI itself, the suite of [SMI Conformance tests](https://layer5.io/projects/service-mesh-interface-conformance) is versioned, and with each new version of SMI, as interfaces added and specifications changed, the Conformance requirements will change as appropriate. The SMI community is the change controller and overseers of what it means to conform to SMI. Work on the mechanics of the conformance tests occurs in the [CNCF Service Mesh WG](https://github.com/cncf/sig-network/tree/master/service-mesh-wg) develops the process and policy around the conformance program.
+
+## Validating Conformance
+
+Conformance validation is performed through automated provisioning of individual service meshes, deployment of a common workload, and generation of service request load as necessary. To facilitate a common set of tests, a simple, instrumented, sample application has been developed for purposes of providing a consistent workload to apply SMI specs against. Deployment of the sample application has been fitted to each service mesh.
+
+### Conformance Test Execution Flow
+
+The following list highlights the sequence of steps taken to perform conformance testing of one type of service mesh.
+
+- **Preconditions**
+  - A given service mesh’s ability to adhere to the SMI specification is validated by running a workload on top of the service mesh.
```

Comment:
Can we test with adaptors? and/or https://github.com/servicemeshinterface/smi-controller-sdk? Basically I want to ensure that we don't give a false impression of what can "work" if certain meshes _could_ work, given the correct adaptor.

Reply:
If I get your meaning correctly, no, it's not feasible this sort of mistake to be made. Yes, Meshery can (does) verify an adapter implementation like that of the SMI adapter for Istio, but ultimately, the tests are end-to-end behavioral tests, which prevent false positives in this way.

Reply:
Ah, I think my question was even more elementary. Imagine I'm looking at a chart with Istio having all red X saying "nope". Is there a way to toggle on the SMI adaptor for Istio and see it turn green?

Reply:
If there isn't a simple way to explain that here, it's okay, but I think that will be a question people have when they read this blog post.

Reply:
Oh, I gotcha. The short answer is, no, there isn't a list of raw capabilities of what each implement _can_ do (or what it says it _can_ do). Consequently, the conformance tool doesn't account for the difference between what it is capable of (what it can do) and what it actually did during the test.

Good of you to inquire about this subject. It is a topic that I've called attention to a fews times, asking for perspectives on Conformance vs Capability. More perspective on the subject - https://docs.google.com/document/d/1HL8Sk7NSLLj-9PRqoHYVIGyU6fZxUQFotrxbmfFtjwc/edit#heading=h.fcpmwjn0roo (engage at-will).